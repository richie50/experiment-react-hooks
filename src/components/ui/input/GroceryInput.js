import "./styles.scss";

import PropTypes from "prop-types";
import React from "react";

const Input = ({ label, type, name, changeHandler, keyPressHandle }, ref) => {
  return (
    <div className="form-field">
      <label htmlFor={name}>{label}</label>
      <input
        data-testid="generic-input"
        ref={ref}
        id={label}
        type={type}
        name={name}
        placeholder={label}
        onChange={changeHandler}
        onKeyDown={keyPressHandle}
      />
    </div>
  );
};

// This basically makes it easy to pass external ref to this component
// For example if you dont want to handle input logic internally
const GroceryInput = React.forwardRef(Input);
// console.log(GroceryInput);

GroceryInput.propTypes = {
  label: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
};

export default GroceryInput;
