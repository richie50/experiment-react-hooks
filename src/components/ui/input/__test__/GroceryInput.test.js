import { render, screen } from "@testing-library/react";

import GroceryInput from "components/ui/input/GroceryInput";

describe("Should match", () => {
  test("snapshot", () => {
    const { container } = render(
      <GroceryInput type="text" label="Snapshot" name="test"></GroceryInput>
    );
    expect(container).toMatchSnapshot();
  });
});

describe("Custom input should be able to render", () => {
  test("text input", () => {
    const { container, unmount } = render(<GroceryInput type={"text"} />);
    const inputElement = screen.getByTestId("generic-input");

    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute("type", "text");
    unmount(container);
  });
  test("password input", () => {
    const { container, unmount } = render(<GroceryInput type={"password"} />);
    const inputElement = screen.getByTestId("generic-input");

    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute("type", "password");
    unmount(container);
  });
  test("number input", () => {
    const { container, unmount } = render(<GroceryInput type={"number"} />);
    const inputElement = screen.getByTestId("generic-input");

    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute("type", "number");
    unmount(container);
  });
});
