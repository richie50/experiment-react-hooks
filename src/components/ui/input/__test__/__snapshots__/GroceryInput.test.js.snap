// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div>
  <div
    class="form-field"
  >
    <label
      for="test"
    >
      Snapshot
    </label>
    <input
      data-testid="generic-input"
      id="Snapshot"
      name="test"
      placeholder="Snapshot"
      type="text"
    />
  </div>
</div>
`;
